yt/frontends/parthenon/data_structures.py

```diff
@@ -0,0 +1,326 @@
+            unit_system=unit_system,
+            default_species_fields=default_species_fields,
+            axis_order=axis_order,
+        )
+        if storage_filename is None:
+            storage_filename = self.basename + ".yt"
+        self.storage_filename = storage_filename
+
+    def _set_code_unit_attributes(self):
+        """
+        Generates the conversion to various physical _units based on the
+        parameter file
+        """
+        for unit, cgs in [
+            ("length", "cm"),
+            ("time", "s"),
+            ("mass", "g"),
+        ]:
+            unit_param = f"Hydro/code_{unit}_cgs"
+            # We set these to cgs for now, but they may have been overridden
+            if getattr(self, unit + "_unit", None) is not None:
+                continue
+            elif unit_param in self.parameters:
+                setattr(
+                    self, f"{unit}_unit", self.quan(self.parameters[unit_param], cgs)
+                )
+            else:
+                mylog.warning(f"Assuming 1.0 = 1.0 {cgs}")
+                setattr(self, f"{unit}_unit", self.quan(1.0, cgs))
+
+        self.magnetic_unit = np.sqrt(
+            self._magnetic_factor
+            * self.mass_unit
+            / (self.time_unit**2 * self.length_unit)
+        )
+        self.magnetic_unit.convert_to_units("gauss")
+        self.velocity_unit = self.length_unit / self.time_unit
+
+    def _parse_parameter_file(self):
+        self.parameters.update(self.specified_parameters)
+        for key, val in self._handle["Params"].attrs.items():
+            if key in self.parameters.keys():
+                mylog.warning(
+                    f"Overriding existing 'f{key}' key in ds.parameters from data 'Params'"
+                )
+            self.parameters[key] = val
+
+        xmin, xmax = self._handle["Info"].attrs["RootGridDomain"][0:2]
+        ymin, ymax = self._handle["Info"].attrs["RootGridDomain"][3:5]
+        zmin, zmax = self._handle["Info"].attrs["RootGridDomain"][6:8]
+
+        self.domain_left_edge = np.array([xmin, ymin, zmin], dtype="float64")
+        self.domain_right_edge = np.array([xmax, ymax, zmax], dtype="float64")
+
+        self.domain_width = self.domain_right_edge - self.domain_left_edge
+        self.domain_dimensions = self._handle["Info"].attrs["RootGridSize"]
+
+        self._field_map = {}
+        k = 0
+
+        dnames = self._handle["Info"].attrs["OutputDatasetNames"]
+        num_components = self._handle["Info"].attrs["NumComponents"]
+
+        if "OutputFormatVersion" in self._handle["Info"].attrs.keys():
+            self.output_format_version = self._handle["Info"].attrs[
+                "OutputFormatVersion"
+            ]
+        else:
+            self.output_format_version = -1
+
+        # Use duck typing to check if num_components is a single int or a list
+        # h5py will return different types if there is a single variable or if there are more
```

Comment:
sounds like we could just use `isinstance(num_components, list)` and drop the try block as well as the comment. It would also improve type safety internally. Or maybe I'm missing something ?

Reply:
yes, that should do the trick.